src/Axiom/Set/Map.agda

```diff
@@ -168,6 +170,14 @@ module Restrictionᵐ (sp-∈ : spec-∈ A) where
   infix 30 _⦅_,-⦆
   _⦅_,-⦆ = curryᵐ
 
+  update : A → Maybe B → Map A B → Map A B
+  update x (just y) m = m ∪ᵐˡ (❴ x , y ❵ , helper)
```

Comment:
There's also a function that constructs a singleton map, so you don't need `helper` at all